pkg/console/subresource/configmap/tech_preview_test.go

```
@@ -0,0 +1,122 @@
+					Name: "cluster",
+				},
+				Spec: operatorv1.ConsoleSpec{},
+			}
+
+			consoleConfig := &configv1.Console{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+			}
+
+			authConfig := &configv1.Authentication{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+			}
+
+			infrastructureConfig := &configv1.Infrastructure{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Status: configv1.InfrastructureStatus{
+					APIServerURL: "https://api.test.cluster:6443",
+				},
+			}
+
+			route := &routev1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "console",
+				},
+				Spec: routev1.RouteSpec{
+					Host: "console.test.cluster",
+				},
+			}
+
+			// Generate configmap with tech preview setting
+			cm, _, err := DefaultConfigMap(
+				operatorConfig,
+				consoleConfig,
+				authConfig,
+				&corev1.ConfigMap{},
+				&corev1.ConfigMap{},
+				infrastructureConfig,
+				route,
+				0,                            // inactivityTimeoutSeconds
+				[]*consolev1.ConsolePlugin{}, // availablePlugins
+				[]string{"amd64"},            // nodeArchitectures
+				[]string{"linux"},            // nodeOperatingSystems
+				false,                        // copiedCSVsDisabled
+				false,                        // contentSecurityPolicyEnabled
+				map[string]string{},          // telemetryConfig
+				"console.test.cluster",       // consoleHost
+				tt.args.techPreviewEnabled,
+			)
+
+			if err != nil {
+				t.Errorf("DefaultConfigMap() error = %v", err)
+				return
+			}
+
+			// Parse the generated config
+			var config consoleserver.Config
+			err = yaml.Unmarshal([]byte(cm.Data["console-config.yaml"]), &config)
+			if err != nil {
+				t.Errorf("Failed to unmarshal config: %v", err)
+				return
+			}
+
+			// Verify tech preview setting
+			if config.ClusterInfo.TechPreviewEnabled != tt.want {
+				t.Errorf("TechPreviewEnabled = %v, want %v", config.ClusterInfo.TechPreviewEnabled, tt.want)
```

suggested change: 

t.Errorf("TechPreviewEnabled: got %t, want %t (case %q, techPreviewEnabled input=%t)",
    config.ClusterInfo.TechPreviewEnabled, tt.want, tt.name, tt.args.techPreviewEnabled)

also, do we want this test case to stop on mismatch? if so, suggest a fatal variant? thinking out loud here. :nerd_face:

---

We don't usually do fatal test cases because they hide other possible failures. We keep our test cases idempotent to prevent them from contaminating each other. So even when a tc fails, the rest _should_ complete normally. I'll update this failure message though.